AgentDiscoveries-Backend/src/main/resources/db/migration/V1__add_initial_tables.sql

```diff
@@ -8,13 +7,14 @@ create table user (
 ) ENGINE=InnoDB DEFAULT CHARSET=utf8;
 
 create table agent (
-  `agent_id` int NOT NULL AUTO_INCREMENT,
+    `call_sign` varchar(20) NOT NULL,
+    `user_id` int NOT NULL,

```

Comment:
So to clarify, we have two related entities:

Users - used to log in to the website.
- Primary key: user_id

Agents - used to file a report, must be linked to a specific user.
- Primary key: call_sign
- Foreign key: user_id onto user.user_id

Is this relationship one-to-one, or are you allowed to have multiple agents linked to a user (or vice-versa)? If you want it to be one-to-one (and I suspect you do), you should have a uniqueness constraint on user_id.

(Unless any of this stuff is a deliberate bug...?)

Reply:
That sounds right; can certainly document it.
I'm assuming the relationship would be at most one-to-one (terminology?) i.e. a user can have at most one agent linked, though it could have none. And an agent must have precisely one user. 
Will add those constraints

Reply:
Yep.  There are many different possible entity relationships

Parent | Child | Suggested Implementation
------------ | ------------- | ------------|
One | One | Use the same table!
One | One or None | Foreign key from child to parent, Unique and Not Null
One or None | One or None | Foreign key from child to parent, Unique and Nullable
One | Many | Foreign key from child to parent, Not Null
One or None | Many | Foreign key from child to parent, Nullable
Many | Many | Join table, combined primary key (assuming no duplicates)

(Usually just referred to as 'one-to-one', 'one-to-many', 'many-to-many' - the uniqueness is kinda separate)